### PR TITLE
Fix argtopk with uneven chunks

### LINKED ISSUE #3719: argtopk broken with uneven chunks
```
>>> a = da.ones(9, chunks=((4, 5), ))
>>> a.argtopk(2)
ValueError: Shapes do not align {0: {2, 3}}
```

The cause of the error is trivial; just need to change https://github.com/dask/dask/blob/669cde6b1a02e5fc7f0bcb924489c6ae8e697323/dask/array/reductions.py#L904 
to ``chunks=(a.chunks[axis], )``.

What baffles me however is the underlying reason why this went undetected so far. Can anybody tell me if there is any good use case for why this should not raise ValueError?!?
```
>>> normalize_chunks((4, 5), shape=(9, ))
((4, 4, 1),)
```

### PR DESCRIPTION
**Pull request by crusaderky**
Closes #3719.

The PR is complete for the specific trivial problem. However I'd also like to change normalize_chunks to prevent similar problems in the future - see discussion in #3719.

### CODE REVIEW

**Comment by crusaderky** [comment]
Same problem here as in the other build - https://travis-ci.org/dask/dask/jobs/400540686#L1337-L1352

**Comment by mrocklin** [comment]
The CI failure has been patched up for now.  I believe that you may have to merge with master though.

File: dask/array/core.py [hunk 1]
```diff
@@ -2090,11 +2095,14 @@ def normalize_chunks(chunks, shape=None, limit=None, dtype=None,
     if not chunks and shape and all(s == 0 for s in shape):
         chunks = ((0,),) * len(shape)
 
+    if (shape and len(shape) == 1 and len(chunks) > 1 and
+            all(type(c) is not tuple for c in chunks)):
```
Comment (mrocklin) [inline dask/array/core.py]: Would it be ok to replace `type(c) is not tuple` with `isinstance(c, numbers.Number)`?  That way we'll probably continue doing something sensible if they pass a list like `chunks=[[1], [2], [3], [4]]`.  Obviously this isn't recommended, but new users can do all sorts of things :)

File: dask/array/tests/test_array_core.py [hunk 2]
```diff
@@ -2226,6 +2226,23 @@ def test_from_array_with_missing_chunks():
...
+    assert normalize_chunks(10, (30, 5)), ((10, 10, 10), (5,))
+    assert normalize_chunks((), (0, 0)), ((0,), (0,))
+    assert normalize_chunks(-1, (0, 3)), ((0,), (3,))
+    assert normalize_chunks("auto", shape=(20,), limit=5, dtype='uint8') == \
+        ((5, 5, 5, 5),)
+
+    with pytest.raises(ValueError):
+        normalize_chunks(((10,), ), (11, ))
+    with pytest.raises(ValueError):
+        normalize_chunks(((5, ), (5, )), (5, ))
```
Comment (mrocklin) [inline dask/array/tests/test_array_core.py]: Thanks for the extra tests here

File: dask/array/tests/test_fft.py [hunk 3]
```diff
@@ -9,9 +9,7 @@
 from dask.array.fft import fft_wrap
 from dask.array.utils import assert_eq, same_keys
 
-from dask.array.core import (
-    normalize_chunks as _normalize_chunks,
-)
+from dask.array.core import normalize_chunks
```
Comment (mrocklin) [inline dask/array/tests/test_fft.py]: Yeah, that makes sense to me.  Thanks

**Comment by mrocklin** [comment]
This seems good to me.  I had one minor comment, but would be happy ignoring it as well.

File: dask/array/core.py (hunk 1 above)
Comment (crusaderky) [reply dask/array/core.py]: ``[[1], [2], [3], [4]]`` is  already covered by the unit tests. Anyway I changed it.

**Comment by crusaderky** [comment]
@mrocklin ready for merge

**Comment by mrocklin** [comment]
Thanks @crusaderky !  Merging.

**Pull request merged**